finesse/hardware/data_file_writer.py

```diff
@@ -35,7 +35,7 @@ def _get_metadata(filename: str) -> dict[str, Any]:
         logging.warning("No time device connected. Using system time.")
         time = datetime.now()
     else:
-        time = datetime.fromtimestamp(time_device.get_time())
```

Comment:
This is a bit confusing. Above, if there is a problem getting the time, the epoch is returned. But here, if there's a problem, `now` is returned. This inconsistency might cause problem. I am not sure what the answer should be, but I would choose one and stick to it consistently across the code.